mm2src/mm2_main/src/lp_swap/maker_swap_v2.rs

```diff
@@ -1,58 +1,93 @@
 use super::{NEGOTIATE_SEND_INTERVAL, NEGOTIATION_TIMEOUT_SEC};
+use crate::mm2::database::my_swaps::{get_swap_events, insert_new_swap_v2, set_swap_is_finished, update_swap_events};
 use crate::mm2::lp_network::subscribe_to_topic;
 use crate::mm2::lp_swap::swap_v2_pb::*;
 use crate::mm2::lp_swap::{broadcast_swap_v2_msg_every, check_balance_for_maker_swap, recv_swap_v2_msg, SecretHashAlgo,
-                          SwapConfirmationsSettings, SwapsContext, TransactionIdentifier};
+                          SwapConfirmationsSettings, SwapsContext, TransactionIdentifier, MAKER_SWAP_V2_TYPE,
+                          MAX_STARTED_AT_DIFF};
 use async_trait::async_trait;
 use bitcrypto::{dhash160, sha256};
-use coins::{ConfirmPaymentInput, FeeApproxStage, GenTakerPaymentSpendArgs, MarketCoinOps, MmCoin, SendPaymentArgs,
-            SwapOpsV2, TxPreimageWithSig};
+use coins::{CoinAssocTypes, ConfirmPaymentInput, FeeApproxStage, GenTakerFundingSpendArgs, GenTakerPaymentSpendArgs,
+            MarketCoinOps, MmCoin, SendPaymentArgs, SwapOpsV2, ToBytes, Transaction, TxPreimageWithSig,
+            ValidateTakerFundingArgs};
 use common::log::{debug, info, warn};
 use common::{bits256, Future01CompatExt, DEX_FEE_ADDR_RAW_PUBKEY};
+use db_common::sqlite::rusqlite::named_params;
 use keys::KeyPair;
 use mm2_core::mm_ctx::MmArc;
+use mm2_err_handle::prelude::*;
 use mm2_number::MmNumber;
 use mm2_state_machine::prelude::*;
 use mm2_state_machine::storable_state_machine::*;
 use primitives::hash::H256;
-use std::collections::HashMap;
+use rpc::v1::types::Bytes as BytesJson;
 use std::marker::PhantomData;
 use uuid::Uuid;
 
 // This is needed to have Debug on messages
 #[allow(unused_imports)] use prost::Message;
 
+/// Negotiation data representation to be stored in DB.
+#[derive(Debug, Deserialize, Serialize)]
+pub struct StoredNegotiationData {
```

Comment:
Here

Reply:
These structs have fields with different semantics/names. Possibly, they will differ even more in the future.

Reply:
I dont insist, just was curious and wanted to note it for the future.

Reply:
> it will be comfortable to use them with different names

I agree, but I had to make them and `Maker/TakerSwapEvent` public since they are used in the public interface here:
https://github.com/KomodoPlatform/komodo-defi-framework/blob/9de1ea29ffca580910fa4ea7bf16194c3577f5c2/mm2src/mm2_main/src/lp_swap/maker_swap_v2.rs#L127

I will look for a way to remove `pub` in next iterations, added to the issue checklist, thanks!